### PR TITLE
Do not shadow variables in src/wallet

### PR DESCRIPTION
**Pull request by paveljanik**
Continues #8105.

Review of this should be pretty straight forward. Just compare binaries. View the diff in visual diff or so.

To provide tested ACK, cherrypick paveljanik@5d2f98e and compare master build log and this PR build log.

### CODE REVIEW

File: src/wallet/db.cpp [hunk 1]
```diff
@@ -387,7 +387,7 @@ bool CDB::Rewrite(const string& strFile, const char* pszSkip)
                         while (fSuccess) {
                             CDataStream ssKey(SER_DISK, CLIENT_VERSION);
                             CDataStream ssValue(SER_DISK, CLIENT_VERSION);
-                            int ret = db.ReadAtCursor(pcursor, ssKey, ssValue, DB_NEXT);
+                            ret = db.ReadAtCursor(pcursor, ssKey, ssValue, DB_NEXT);
```
Comment (laanwj) [inline src/wallet/db.cpp]: I'm not sure promoting these return values up to the function scope is a good idea. Keeping the scope of variables as small as possible ensures that results won't leak or accidentally influence other things.

Comment (paveljanik) [inline src/wallet/db.cpp]: Technically speaking, this is not directly function scope.

But yes, one view is to have only one return value variable, the other is one return value variable per function call.

Renaming `ret` to `ret1` here could make this particular change smaller.

Comment (laanwj) [inline src/wallet/db.cpp]: Yes, I'd prefer just renaming

Comment (paveljanik) [inline src/wallet/db.cpp]: Done and squashed - now all the changes are of the same kind.

**Comment by MarcoFalke** [comment]
ACK: `objdump -d $bin` returns the same binaries for me on d5b25fa

**Comment by paveljanik** [comment]
more ACKs please

**Comment by sipa** [comment]
utACK d5b25fa5792b3899759a73059a87c6b015fc8535

**Comment by paveljanik** [comment]
Rebased.

**Comment by laanwj** [comment]
Also compared binaries, tACK b175cb7

**Pull request merged**